### ISSUE TITLE
Correlation topic doesn't persist on edit

### ISSUE THREAD

**Issue by tito** [dxdc/node-red-contrib-join-wait#8]
Correlation topic doesn't persist on edit. To reproduce the issue:

- Edit the join-wait node
- Write something in the Correlation Topic (msg.payload.xxx)
- Click on Done
- Edit the join-wait node
- -> Correlation Topic is empty

**Comment by dxdc** [comment]
Thanks for this report @tito. I experimented with it a bit, and I think it must be a bug in the editor somewhere which was introduced in NR v2. The node actually saves the correlationTopic and it also works properly, it just doesn't show in the editor window. So whatever you enter here will still work - and show - if you export the node.

If you look at `join-wait.html`, removing `hasValue: false` "fixes" the issue, but it makes the behavior of the undefined selection not ideal. I think it's probably a bug in the NR editor but there may be another way to fix it. I've cross-posted the problem on the NR forums for assistance.

```js
            $('#node-input-correlationTopic').typedInput({
                default: this.correlationTopicType || types.Undefined,
                typeField: $('#node-input-correlationTopicType'),
                types: [
                    { value: 'undefined', label: RED._('common.type.undefined'), hasValue: false },
                    'msg',
                    'flow',
                    'global',
                    'jsonata',
                ],
            });
            ```

**Comment by dxdc** [comment]
Nick O'Leary (lead Node-Red developer/creator) already responded and said:

> You have found a bug. I have a fix and we'll publish it in 2.1.4 later this week.

Closing the issue as it will be fixed as part of core. Thanks for the report!
